browser/net/ipfs_redirect_network_delegate_helper.cc

```diff
@@ -8,15 +8,18 @@
 #include <string>
 
 #include "brave/components/ipfs/ipfs_utils.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
 #include "net/base/net_errors.h"
 
 namespace ipfs {
 
 int OnBeforeURLRequest_IPFSRedirectWork(
     const brave::ResponseCallback& next_callback,
     std::shared_ptr<brave::BraveRequestInfo> ctx) {
-  if (!ctx->browser_context ||
-      IsIpfsResolveMethodDisabled(ctx->browser_context)) {
+  auto* prefs = user_prefs::UserPrefs::Get(ctx->browser_context);
+  if (!ctx->browser_context || IsIpfsResolveMethodDisabled(prefs)) {
```

Comment:
we'd better check `ctx->browser_context` before doing `UserPrefs::Get(ctx->browser_context)`